plugins/detabler/plugin.js

```diff
@@ -1,9 +1,69 @@
 'use strict';
 
-(function() {
-    CKEDITOR.plugins.add( 'detabler', {    
-    init: function( editor ) {
-        // Plugin logic goes here...
-    }
-});
-})();
\ No newline at end of file
+(function () {
+    CKEDITOR.plugins.add('detabler', {
+		requires: 'table,contextmenu',
+		init: function (editor) {
+			editor.addCommand('detable', {
+				exec: function (editor) {
+					var path,
+						table,
+						cells,
+						currentCell,
+						currentItem;
+
+					path = editor.elementPath();
+					table = path.contains('table');
+
+					if (table) {
+						table.$.normalize(); // we don't want to deal with un-merged text nodes here.
+						cells = table.find('td, th');
+						// loop through all cell-like items
+						for (var i = 0; i < cells.count(); i++) {
+							currentCell = cells.getItem(i);
+
+							// we can't iterate through the children nodeList here
+							// because this while loop removes items from children 1 by 1
+							while (currentItem = currentCell.getChild(0)) {
```

Comment:
Yup, null:  http://docs.ckeditor.com/#!/api/CKEDITOR.dom.element-method-getChild

Does the comment make sense?

Basically we can't for loop through the children of a table cell, because we remove those children from the cell as part of this operation.  (actually, this is probably a better comment than what i actually wrote.)